src/status_im2/contexts/quo_preview/wallet/network_routing.cljs

```diff
@@ -0,0 +1,93 @@
+(ns status-im2.contexts.quo-preview.wallet.network-routing
+  (:require [quo2.core :as quo]
+            [react-native.core :as rn]
+            [reagent.core :as reagent]
+            [status-im2.contexts.quo-preview.preview :as preview]))
+
+(def descriptor
+  [{:label   "Number of networks"
+    :key     :number-networks
+    :type    :select
+    :options [{:key 2} {:key 3} {:key 4} {:key 5}]}])
+
+(defn- fake-call-to-get-amounts
+  [{:keys [new-amount fixed-index current-values on-success]}]
+  (let [number-networks        (count current-values)
+        amount-difference      (- (get current-values fixed-index) new-amount)
+        difference-distributed (/ amount-difference (dec number-networks))
+        new-values             (assoc (mapv #(+ % difference-distributed) current-values)
+                                      fixed-index
+                                      new-amount)]
+    (js/setTimeout #(on-success new-values) (rand-nth (range 700 5000 250)))))
+
+(defn preview-internal
+  [{:keys [total-amount number-networks] :as descriptor-state}]
+  (let [initial-amount   (/ total-amount number-networks)
+        networks         (reagent/atom
```

Comment:
I don't know where the problem is, but if I extract `networks` to a global state and mutate it in the REPL, the bars are not re-rendered correctly. I did this to simulate external changes to the prop `networks`.

```clojure
;; Let's say I start with
(def networks
  (reagent/atom
   [{:amount       50
     :max-amount   300
     :network-name :xDai}
    {:amount       80
     :max-amount   350
     :network-name :ethereum}
    {:amount       120
     :max-amount   350
     :network-name :arbitrum}
    {:amount       150
     :max-amount   250
     :network-name :optimism}]))

;; Then in the REPL, to simulate an external change to the `networks` I changed the order and some values.
(reset! networks
        [{:amount       120
          :max-amount   350
          :network-name :arbitrum}
         {:amount       90
          :max-amount   300
          :network-name :xDai}
         {:amount       40
          :max-amount   350
          :network-name :ethereum}
         {:amount       150
          :max-amount   250
          :network-name :optimism}])
```

The colors of the bars change, but not their order or values.

Reply:
Explanation in the comment below